File: 02Algorithm_Study/01_week_homeWork/Big_number.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;



bool compare(const string& a,const string& b){
    /*
    int len = min(a.size(),b.size());
    
    if(a.size() == b.size()){
        if(a[0] == b[0]){
        int tmpa = stoi(a);
        int tmpb = stoi(b);
        return tmpa>tmpb;
    }
        else{
            return a[0] > b[0];
        }
    }
    else{
        
    }
    */
    return a +b > b + a;
        
}
string solution(vector<int> numbers) {
    string answer = "";
    vector <string> vecstr;
    
    int len = numbers.size();
    
    for (int i = 0; i < len; i ++){
        string tmp = to_string(numbers[i]);
        vecstr.push_back(tmp);        
    }
    sort(vecstr.begin(),vecstr.end(),compare);
    
    for(int i = 0; i < vecstr.size(); i ++){
        answer += vecstr[i];
    }
    if(answer[0] == '0'){
        answer = '0';
    }
    
    return answer;
}

File: 02Algorithm_Study/01_week_homeWork/Big_number.py
from functools import cmp_to_key
def compare(a : str,b : str):
    return a + b > b + a


def solution(numbers):
    answer = ''
    sortList = [str(i) for i in numbers]
    
    
    sortList.sort(key=cmp_to_key(lambda x, y : 1 if x + y > y + x else -1))
    #print(sortList)
    sortList.reverse()
    #sortList = sorted(sortList,key=cmp_to_key(compare))
    #print(sortList)
    if(sortList[0] =='0'):
        answer = '0'
    else:
        for i in sortList:
            answer +=i
    
    return answer